### PR TITLE
Upgrade go-boost-utils

### PR DESCRIPTION
**Pull request by avalonche**
## 📝 Summary

Upgrade relay code to the latest go-boost-utils as the library migrated to attestant types.

## 📚 References

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [ ] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

**Review by jtraglia — CHANGES_REQUESTED** [review]
Here's my initial review. Overall looks pretty good.

File: services/api/optimistic_test.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const (
 	slot        = uint64(41)
 	collateral  = 1000
 	builderID   = "builder0x69"
-	randao      = "01234567890123456789012345678901"
+	randao      = "0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2"
```
[the remark this answers is not in the document]
Comment (avalonche) [reply services/api/optimistic_test.go:33]: `01234567890123456789012345678901` had an invalid length, `0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2` is one of the example values from the [beacon-API](https://github.com/ethereum/beacon-APIs/blob/20980bf18dc71306c9157a0b7f82c164bf84f76d/types/primitive.yaml#L70)

File: services/api/types_test.go [hunk 2]
```diff
@@ -3,45 +3,52 @@ package api
 import (
 	"testing"
 
-	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/attestantio/go-builder-client/api/capella"
+	apiv1 "github.com/attestantio/go-builder-client/api/v1"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	capellaspec "github.com/attestantio/go-eth2-client/spec/capella"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
```
[the remark this answers is not in the document]
Comment (avalonche) [reply services/api/types_test.go:10]: I'll raise a new pr for the renaming as this cuts across more than changes to go-boost-utils

File: services/api/optimistic_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jtraglia) [reply services/api/optimistic_test.go:33]: Gotcha. Thanks.

File: services/api/types_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jtraglia) [reply services/api/types_test.go:10]: Okay. Sounds good. Works for me.

File: go.mod
[the remark this answers is not in the document]
Comment (jtraglia) [reply go.mod:1]: Ah I understand. Is the plan to use your fork until that PR has been merged & released? Or would you prefer we wait until it's upstreamed to merge this PR?

File: services/api/service.go [hunk 3]
```diff
@@ -933,7 +935,7 @@ func (api *RelayAPI) handleRegisterValidator(w http.ResponseWriter, req *http.Re
 			return nil, fmt.Errorf("registration message error (timestamp): %w", err)
 		}
 
-		timestamp, err := strconv.ParseUint(_timestamp, 10, 64)
+		timestamp, err := strconv.ParseInt(_timestamp, 10, 64)
```
[the remark this answers is not in the document]
Comment (jtraglia) [reply services/api/service.go:939]: Thanks 👍

File: go.mod
[the remark this answers is not in the document]
Comment (avalonche) [reply go.mod:1]: This PR will be merged into the `develop` so it'll be more a WIP branch using my fork until the changes in `go-builder-client` stabilise. There are a few more changes needed in that library.

File: services/api/service_test.go [hunk 4]
```diff
@@ -245,26 +245,26 @@ func TestRegisterValidator(t *testing.T) {
...
+	// 	payload, err := generateSignedValidatorRegistration(nil, bellatrix.ExecutionAddress{1}, td+10)
 	// 	require.NoError(t, err)
 	// 	err = backend.redis.SetKnownValidator(payload.Message.Pubkey.PubkeyHex(), 1)
 	// 	require.NoError(t, err)
 	// 	_, err = backend.datastore.RefreshKnownValidators()
 	// 	require.NoError(t, err)
 
-	// 	rr := backend.request(http.MethodPost, path, []types.SignedValidatorRegistration{*payload})
+	// 	rr := backend.request(http.MethodPost, path, []apiv1.SignedValidatorRegistration{*payload})
 	// 	require.Equal(t, http.StatusOK, rr.Code, rr.Body.String())
```
[the remark this answers is not in the document]
Comment (avalonche) [reply services/api/service_test.go]: everything should be cleaned up now

**Review by jtraglia — APPROVED** [review]
LGTM 👍 great work!

**Pull request merged**